components/service-operator/apis/storage/v1beta1/image_repository_types.go

```diff
@@ -174,3 +183,52 @@ func (s *ImageRepository) GetStackRoleParameters(roleName string) ([]*cloudforma
 	}
 	return params, nil
 }
+
+func (s *ImageRepository) Empty(ctx context.Context, client sdk.Client) error {
+	var imageIdBatches [][]*ecr.ImageIdentifier
+	err := client.DescribeImagesPagesWithContext(
+		ctx,
+		&ecr.DescribeImagesInput{
+			RepositoryName: aws.String(s.GetAWSName()),
+		},
+		func(images *ecr.DescribeImagesOutput, _ bool) bool {
+			var imageIds []*ecr.ImageIdentifier
+			for _, image := range images.ImageDetails {
+				imageIds = append(imageIds, &ecr.ImageIdentifier{
+					ImageDigest: image.ImageDigest,
+				})
+			}
+			if len(imageIds) > 0 {
+				imageIdBatches = append(imageIdBatches, imageIds)
+			}
+			return true
+		},
+	)
+
+	if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
This block needs to be inside of the `if err != nil ` block.

for reasons that may hurt your brain, it is possible for `err` to be `nil` AND pass as `ok` for that type assertion.